src/instructlab/model/list.py

```diff
@@ -0,0 +1,132 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+import logging
+import os
+import time
+
+# Third Party
+from huggingface_hub import ModelCard
+from requests import exceptions as requests_exceptions
+import click
+import pandas
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+from instructlab.model.backends.backends import is_model_gguf
+
+logger = logging.getLogger(__name__)
+
+
+@click.command(name="list")
+@clickext.display_params
+@click.option(
+    "--model-dirs",
+    help="Base directories where models are stored.",
+    multiple=True,
+    default=[DEFAULTS.MODELS_DIR],
+    show_default=True,
+)
+@click.option(
+    "--list-checkpoints",
+    type=click.BOOL,
+    default=False,
+    is_flag=True,
+    show_default=True,
+)
+def model_list(model_dirs: list[str], list_checkpoints: bool):
+    """lists models"""
+    data = []
+    for directory in model_dirs:
+        for entry in Path(directory).iterdir():
+            # if file, just tally the size. This must be a GGUF.
+            if entry.is_file() and is_model_gguf(entry):
+                # stat the gguf, add it to the table
+                stat = Path(entry.absolute()).stat(follow_symlinks=False)
+                f_size = stat.st_size
+                f_size, magnitude = convert_bytes_to_proper_mag(f_size)
+                # add to table
+                age = pandas.to_timedelta(time.time_ns() - stat.st_ctime_ns)
+                data.append([entry.name, str(age), f"{f_size} {magnitude}"])
+            elif entry.is_dir():
+                actual_model_name = ""
+                all_files_sizes = 0
+                add_model = False
+                # walk entire dir.
+                for root, _, files in os.walk(entry.as_posix()):
+                    normalized_path = os.path.normpath(root)
```

Comment:
pathlib objects have a `walk` method and `resolve` method. The following change should simplify your code. Then you can replace `parts = normalized_path.split(os.sep)` with `root.parts`.

```suggestion
                for root, _, files in entry.resolve().walk():
```

Reply:
thanks for the tip! will change accordingly

Reply:
getting `AttributeError: 'PosixPath' object has no attribute 'walk'`

Reply:
argh, my bad. The method was added in 3.12. We have to support 3.10 and 3.11.